docs/spec/ibc/overview.md

```diff
@@ -0,0 +1,39 @@
+## 1 Overview
+
+([Back to table of contents](specification.md#contents))
+
+The IBC protocol creates a mechanism by which multiple sovereign replicated fault tolerant state machines my pass messages to each other. These messages provide a base layer for the creation of communicating blockchain architecture that overcomes challenges in the scalability and extensibility of computing blockchain environments.
+
+The IBC protocol assumes that multiple applications are running on their own blockchain with their own state and own logic. Communication is achieved over an extremely secure message queue protocol, allowing the creation of complex inter-chain processes without trusted parties. This architecture can be seen as a parallel to microservices in the blockchain space, and the IBC protocol can be seen as an analog to the AMQP messaging protocol[[2](./footnotes.md#2)], used by StormMQ, RabbitMQ, etc.
```

Comment:
I am confused with the notion of queue at this level as I thought IBC is TCP for blockchain apps.

Reply:
Doesn't TCP also have the notion of a queue (You get messages in FIFO order, even if the network delivers them out of order)?

Reply:
Added several sections about the kinds of invariants applications may want to enforce, and how they could enforce them through IBC.